src/v/kafka/server/group_manager.cc

```diff
@@ -717,6 +717,19 @@ group_manager::recover_offsets(group_offsets_snapshot snap) {
         kafka_r.ntp = offsets_ntp;
         auto& kafka_data = kafka_r.data;
         kafka_data.group_id = kafka::group_id(g.group_id);
+        auto group = get_group(kafka_data.group_id);
+        if (group && !group->in_state(group_state::empty)) {
```

Comment:
what about state `dead`?

Reply:
Good question. Right now the `offset_commit` method [will not work](https://github.com/redpanda-data/redpanda/blob/a5781dde6b9af3c7ed0f4f021d248de881efdfac/src/v/kafka/server/group.cc#L2455). I'm not too familiar with how groups are cleaned up -- at what point is it `erase()`d? Naively I'd think we'd just want to remove the group and then run the commit, but I suspect that'll just break

Reply:
Out of band we discussed not overwriting groups if they exist in any state.

Reply:
Done